include/SFML/System/Angle.inl

```diff
@@ -0,0 +1,259 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+namespace priv
+{
+    constexpr float pi = 3.141592654f;
+    constexpr float tau = 2 * pi;
+
+    constexpr float positiveRemainder(float a, float b)
+    {
+        assert(b > 0.0f);
+        if (a >= 0.f)
+            return a - static_cast<float>(static_cast<int>(a / b)) * b;
+        else
+            return a + b - static_cast<float>(static_cast<int>(a / b)) * b;
+    }
```

Comment:
This method doesn't seem to return a remainder in the interval 0 <= r < |b| for negative numbers that divide exactly by b.
This is evident from one of the remainder tests: `CHECK((sf::degrees(-90.0f) % sf::degrees(30.0f)).asDegrees() == Approx(30.0f))`.

```suggestion
    constexpr float positiveRemainder(float a, float b)
    {
        assert(b > 0.0f);
        const float val = a - static_cast<float>(static_cast<int>(a / b)) * b;
        
        if (val >= 0.f)
            return val;
        else
            return val + b;
    }
```

Reply:
Thanks this works! Once I swapped implementation the test case you mentioned started failing. I have this fixed locally and will update the remote branch shortly.